feat: add `stats/incr/nanmvmr`

23f3001304 opened the pull request:
Resolves #5615 .

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the @stdlib/stats/incr/nanmvmr package, which computes moving [variance-to-mean ratio](https://en.wikipedia.org/wiki/Index_of_dispersion) (VMR) incrementally while handling NaN values by ignoring them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5615 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other
<i>Modified test cases are passing<i>

![image](https://github.com/user-attachments/assets/3e489e09-5368-48f5-9c3d-5f89b0c0dbdd)

<i> Examples with NaN generated successfully <i>

![image](https://github.com/user-attachments/assets/0f15fb34-601f-41cc-82d0-b78c45f721ab)

<i> All 9 benchmarks are passed <i>

![image](https://github.com/user-attachments/assets/45746fc6-90fd-41bf-b104-6b446eaf4755)



> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

23f3001304 committed:
Added nanmvmr

23f3001304 committed:
Changed Packagejson

23f3001304 committed:
Updated Readme.md

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

23f3001304 committed:
[Fixed Lint Issues]

23f3001304 committed:
[Fixed Lint Issues]

23f3001304 committed:
[Fixed Issues]

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmvmr/README.md:
```diff
@@ -0,0 +1,227 @@
+
+// For each simulated datum, update the moving variance-to-mean ratio...
+for ( i = 0; i < 100; i++ ) {
+    v = ( randu()> 0.2 ) ? NaN : randu() * 100.0;
```
Missing "ignoring NaN values" here too. Also, missing space before `>` and the NaN probability is inverted — `randu() > 0.2` produces NaN ~80% of the time.

```suggestion
    v = ( randu() < 0.2 ) ? NaN : randu() * 100.0;
```

Planeshifter commented:
/stdlib update-copyright-years

Planeshifter committed:
fix: address review feedback for stats/incr/nanmvmr

- Fix examples/index.js: use if/else pattern for NaN generation and fix
  console.log null check
- Fix README.md: add 'ignoring NaN values' to h1, usage description, and
  accumulator description; add NaN values to usage example; add NaN input
  handling note; fix inverted NaN probability in examples section
- Fix package.json: add 'nan' and 'ignore' keywords
- Fix docs/types/index.d.ts: remove contradictory note about NaN
  propagation; use correct NaN-ignoring note consistent with nanmstdev

Planeshifter committed:
fix: update copyright years and migrate to ULP-based test assertions

Planeshifter committed:
fix: remove pre-populated auto-generated related packages section

Planeshifter committed:
fix: update README examples to print values at each iteration

Planeshifter reviewed (approved):
Let's get it in. Thanks, @23f3001304!
